pkg/sensors/tracing/uprobe_test.go

```diff
@@ -196,6 +198,118 @@ spec:
 	require.NoError(t, err)
 }
 
+func TestUretprobeGeneric(t *testing.T) {
+	testUretprobe := testutils.RepoRootPath("contrib/tester-progs/uretprobe")
+	uretprobeHook := `
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: uretprobe
+spec:
+  uprobes:
+  - path: "` + testUretprobe + `"
+    symbols:
+    - "return_string"
+    return: true
+    returnArg:
+      index: 0
+      type: "string"
+`
+
+	uretprobeConfigHook := []byte(uretprobeHook)
+	err := os.WriteFile(testConfigFile, uretprobeConfigHook, 0644)
+	if err != nil {
+		t.Fatalf("writeFile(%s): err %s", testConfigFile, err)
+	}
+
+	upChecker := ec.NewProcessUprobeChecker("URETPROBE_GENERIC").
+		WithProcess(ec.NewProcessChecker().
+			WithBinary(sm.Full(testUretprobe))).
+		WithSymbol(sm.Full("return_string")).
+		WithArgs(ec.NewKprobeArgumentListMatcher().
+			WithOperator(lc.Ordered).
+			WithValues(
+				ec.NewKprobeArgumentChecker().WithStringArg(sm.Full("ret input")),
+			))
+	checker := ec.NewUnorderedEventChecker(upChecker)
+
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	obs, err := observertesthelper.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib, observertesthelper.WithMyPid())
+	if err != nil {
+		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+	}
+	observertesthelper.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	if err := exec.Command(testUretprobe).Run(); err != nil {
+		t.Fatalf("Failed to execute test binary: %s\n", err)
```

Comment:
fyi: the tests from which you copy pasted this are having issues in vmtests, I think because of panic/or fatal, not sure. you might want to lookup the vmtests results for 5.4 for example even if it's green and switch to more traditional `require.NoError` or `t.Error`.